Tests/MapboxNavigationTests/MapViewTests.swift

```diff
@@ -137,4 +138,151 @@ class MapViewTests: XCTestCase {
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in English after localization.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "es"))
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_es" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in Spanish after localization.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "zh-Hant-TW"))
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_zh-Hant" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Place labels should be in Traditional Chinese after localization.")
+        
+        mapView.localizeLabels(into: Locale(identifier: "zh-Hans-CN"))
+        
+        XCTAssertEqual(textFieldExpression(layerIdentifier: "road-labels"),
+                       Exp(.format) {
+                        Exp(.coalesce) { Exp(.get) { "name_en" }; Exp(.get) { "name" } }
+                        FormatOptions()
+                       },
+                       "Road labels should remain in English after localization.")
+        // https://github.com/mapbox/mapbox-maps-ios/issues/652
+        XCTAssertNotEqual(textFieldExpression(layerIdentifier: "place-labels"),
+                          Exp(.format) {
+                            Exp(.coalesce) { Exp(.get) { "name_zh-Hans" }; Exp(.get) { "name" } }
+                            FormatOptions()
+                          },
+                          "Place labels should be in Simplified Chinese after localization.")
+        
+        expect { mapView.localizeLabels(into: Locale(identifier: "tlh")) }
+            .to(throwAssertion(), description: "Localization into Klingon should not be supported. 🖖")
```

Comment:
The map SDK throws a fatal error rather if we pass in an unsupported Locale: mapbox/mapbox-maps-ios#655. This is problematic because the user could have set the system language preference to some language that the Streets source doesn’t support, triggering a crash. If this issue isn’t fixed upstream, we’ll have to hard-code the same array of supported languages in the navigation SDK, making it even more difficult for the server team to add support for new language.

Reply:
> If this issue isn’t fixed upstream, we’ll have to hard-code the same array of supported languages in the navigation SDK, making it even more difficult for the server team to add support for new language.

I wound up implementing this workaround for now. We can easily remove the workaround once the map SDK is fixed.